File: 10_sql_python/trulogic_prueba_hacker_rank_2025_agosto_8/respuesta_rest_api_get_author_articles.py
import requests

def getArticleTitles(author):
    titles = []
    page = 1
    
    while True:
        # Construir la URL con el autor y número de página
        url = f"https://jsonmock.hackerrank.com/api/articles?author={author}&page={page}"
        response = requests.get(url)
        data = response.json()
        
        # Procesar artículos en esta página
        for article in data['data']:
            if article['title']:
                titles.append(article['title'])
            elif article['story_title']:
                titles.append(article['story_title'])
            # Si ambos son None, se ignora el artículo
        
        # Verificar si ya se llegó a la última página
        if page >= data['total_pages']:
            break
        page += 1
    
    return titles

# Ejemplo de uso
if __name__ == '__main__':
    author = input().strip()
    result = getArticleTitles(author)
    for title in result:
        print(title)

File: sql_python/trulogic_prueba_hacker_rank_2025_agosto_8/respuesta_rest_api_get_author_articles.js
async function getArticleTitles(author) {
    let titles = [];
    let page = 1;
    let totalPages = 1;

    while (page <= totalPages) {
        const url = `https://jsonmock.hackerrank.com/api/articles?author=${author}&page=${page}`;
        const response = await fetch(url);
        const data = await response.json();
        totalPages = data.total_pages;

        for (const article of data.data) {
            if (article.title) {
                titles.push(article.title);
            } else if (article.story_title) {
                titles.push(article.story_title);
            }
        }
        page++;
    }

    return titles;
}
'use strict';

const fs = require('fs');
const axios = require('axios');
const fetch = require('node-fetch');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}



/*
 * Complete the 'getArticleTitles' function below.
 *
 * The function is expected to return a STRING_ARRAY.
 * The function accepts STRING author as parameter.
 * 
 * URL for cut and paste: 
 * https://jsonmock.hackerrank.com/api/articles?author=<authorName>&page=<num>
 *
 */

async function getArticleTitles(author) {
    let titles = [];
    let page = 1;
    let totalPages = 1;

    while (page <= totalPages) {
        const url = `https://jsonmock.hackerrank.com/api/articles?author=${author}&page=${page}`;
        const response = await fetch(url);
        const data = await response.json();
        totalPages = data.total_pages;

        for (const article of data.data) {
            if (article.title) {
                titles.push(article.title);
            } else if (article.story_title) {
                titles.push(article.story_title);
            }
        }
        page++;
    }

    return titles;
}

async function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const author = readLine();

    const result = await getArticleTitles(author);

    ws.write(result.join('\n') + '\n');

    ws.end();
}
